### PR TITLE
CODICE-2 Removed ddf-support modules, migrated common Codice properties

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Removed DDF-Support modules to make this a true parent project. Migrated common properties from downstream projects up to the pom here to be tracked.
#### Who is reviewing it? 
@garrettfreibott @austinsteffes @ahoffer @ricklarsen @shaundmorris @pklinef @clockard @djblue @lessarderic @coyotesqrl 

Please closely inspect the updates in pom.xml included in these changes.

### CODE REVIEW

**Review by coyotesqrl — APPROVED** [review]
I assume a new repo has/is being created for the modules removed from here?

File: pom.xml [hunk 1]
```diff
@@ -33,35 +33,241 @@
...
+        <jackson-core-version>2.9.4</jackson-core-version>
+        <jai-imageio-core.version>1.3.1</jai-imageio-core.version>
+        <JavaAPIforKml.version>2.2.1_CODICE_1</JavaAPIforKml.version>
+        <javax.annotation-api.version>1.3</javax.annotation-api.version>
+        <javax.inject.version>1</javax.inject.version>
+        <javax.servlet-api.version>3.1.0</javax.servlet-api.version>
+        <javax.ws.rs.version>2.1</javax.ws.rs.version>
+        <javax-el.version>3.0.0</javax-el.version>
+        <javax-mail.version>1.5.6</javax-mail.version>
+        <jaxb.version>2.2.11</jaxb.version>
```
Comment (clockard) [inline pom.xml:168]: these version seem backward. Is the 2.2.11 suppose to be the jaxb2.version property?

File: pom.xml [hunk 2]
```diff
@@ -33,35 +33,241 @@
...
+        <commons-validator.version>1.6</commons-validator.version>
+        <components-backbone.version>1.1.0</components-backbone.version>
+        <components-font-awesome.version>4.7.0</components-font-awesome.version>
+        <countryconverter.version>0.1.3</countryconverter.version>
+        <cxf.rt-binding-objects.version>3.1.15</cxf.rt-binding-objects.version>
+        <cxf.rt-databinding-jibx.version>3.1.15</cxf.rt-databinding-jibx.version>
+        <cxf.rt-databinding-xmlbeans.version>3.1.15</cxf.rt-databinding-xmlbeans.version>
+        <cxf.rt-management-web.version>3.1.15</cxf.rt-management-web.version>
+        <cxf.version>3.2.5</cxf.version>
+        <ddf.support.version>2.3.14</ddf.support.version>
```
Comment (clockard) [inline pom.xml]: Should we include ddf support in the parent?

File: pom.xml [hunk 3]
```diff
@@ -33,35 +33,241 @@
...
+        <ddmsence.version>2.6.0</ddmsence.version>
```
Comment (clockard) [inline pom.xml]: Should this be included here?

**Review by clockard — APPROVED** [review]
I like the ordering of the properties. Next step is to normalize property names :)

**Comment by mcalcote** [comment]
@coyotesqrl the modules will still exist in ddf-support. I had overestimated what should be included within a parent project and am focusing on dependency and plugin management first.

File: pom.xml (hunk 3 above)
Comment (mcalcote) [reply pom.xml]: Looks like that dependency doesn't need to be at this level. Removed

File: pom.xml (hunk 2 above)
Comment (mcalcote) [reply pom.xml]: I suppose not, each project can define the version in their own root poms.

File: pom.xml (hunk 1 above)
Comment (mcalcote) [reply pom.xml:168]: This came straight out of DDF https://github.com/codice/ddf/blob/master/pom.xml#L201

**Review by austinsteffes — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -33,35 +33,240 @@
...
-        <module>support-findbugs</module>
-        <module>support-owasp</module>
-        <module>support-pmd</module>
-        <module>support-karaf</module>
-        <module>version-validation-plugin</module>
-    </modules>
-
     <properties>
-        <maven.assembly.plugin.version>2.2.2</maven.assembly.plugin.version>
+        <abdera.osgi.version>1.1.3_2</abdera.osgi.version>
```
Comment (lessarderic) [inline pom.xml]: ❓I like the alphabetical order, but I'm wondering if that's enough. Should we consider keeping the build/maven, documentation and test dependency versions separate?

File: pom.xml [hunk 5]
```diff
@@ -33,35 +33,240 @@
...
+        <pax.exam.version>4.11.0</pax.exam.version>
+        <pax.url.version>2.5.2</pax.url.version>
+        <pax.web.version>6.0.11</pax.web.version>
+        <pdfbox.version>2.0.11</pdfbox.version>
+        <picocontainer.version>1.3_1</picocontainer.version>
+        <plexus-compiler-javac-errorprone.version>2.8.2</plexus-compiler-javac-errorprone.version>
+        <poi.contrib.version>3.7-beta3</poi.contrib.version>
+        <poi.version>3.17</poi.version>
+        <powermock.version>1.6.4</powermock.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (lessarderic) [inline pom.xml]: ❓Should we keep the `project.` properties separate since they are simple string replacements and not versions?

**Review by lessarderic — COMMENTED** [review]
❓Are we planning to add a `Jenkinsfile` for this repo and update the pipelines to trigger all the downstream projects when it changes?

File: pom.xml (hunk 4 above)
Comment (mcalcote) [reply pom.xml]: We can. I would assume most people will use the find in page/search functionality when making edits, but I could see your point about groups of dependencies being edited together. I can give it an attempt.

File: pom.xml (hunk 5 above)
Comment (mcalcote) [reply pom.xml]: Separate in this pom or separate as in the downstream projects?

**Review by lessarderic — APPROVED** [review]

**Pull request merged**